core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -106,7 +106,10 @@ public boolean execute(
             shouldContinue = getEvaluatedCondition(workflow, task, workflowExecutor);
             LOGGER.debug("taskid {} condition evaluated to {}", task.getTaskId(), shouldContinue);
             if (shouldContinue) {
+                // Update the iteration value and put in output in case any of loopover task to
+                // consume.
                 task.setIteration(task.getIteration() + 1);
+                task.getOutputData().put("iteration", task.getIteration());
```

Comment:
will this work if I do ${doWhileTaskRef.output.iteration} from one of the looped task.
I noticed in the current implementation the first task in the loop is scheduled before the doWhile itself so it does not have a reference to the while task's output available.

Reply:
verified it works :)

Reply:
The integration tests have exactly the same thing. https://github.com/orkes-io/conductor/pull/20/files#diff-e5236a16ccd5849f5a19bb6b4939db7f46728e618286028452f5dd4e88ecda3aR27 . It also shows usage of array indexing which start from 0.